File: FrameworkExample/Objective-C/VponSdkSampleObjC/VponSdkSampleObjC/Sdk/VponSdkSplashViewController.m
//
//  VponSdkSplashViewController.m
//  VponSdkSampleObjC
//
//  Created by EricChien on 2017/11/29.
//  Copyright © 2017年 Soul. All rights reserved.
//

#import "VponSdkSplashViewController.h"

@import VpadnSDKAdKit;
@import AdSupport;

@interface VponSdkSplashViewController () <VpadnSplashDelegate>

@property (weak, nonatomic) IBOutlet UIView *loadSplashView;

@property (strong, nonatomic) VpadnSplash *vpadnSplash;

@property (weak, nonatomic) IBOutlet  UIButton *requestButton;

@end

@implementation VponSdkSplashViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"SDK - Splash";
    [self requestButtonDidTouch:_requestButton];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Initial VpadnAdRequest

- (VpadnAdRequest *) initialRequest {
    VpadnAdRequest *request = [[VpadnAdRequest alloc] init];
    [request setTestDevices:@[[ASIdentifierManager sharedManager].advertisingIdentifier.UUIDString]];   //取得測試廣告
    [request setUserInfoGender:VpadnGenderMale];                                                        //性別
    [request setUserInfoBirthdayWithYear:2000 Month:8 andDay:17];                                       //生日
    [request setMaxAdContentRating:VpadnMaxAdContentRatingGeneral];                                     //最高可投放的年齡(分類)限制
    [request setTagForUnderAgeOfConsent:VpadnTagForUnderAgeOfConsentFalse];                             //是否專為特定年齡投放
    [request setTagForChildDirectedTreatment:VpadnTagForChildDirectedTreatmentFalse];                   //是否專為兒童投放
    [request setContentUrl:@"https://www.vpon.com.tw/"];
    [request setContentData:@{@"key1": @(1), @"key2": @(YES), @"key3": @"name", @"key4": @(123.31)}];
    return request;
}

#pragma mark - Button Method

- (IBAction)requestButtonDidTouch:(UIButton *)sender {
    sender.enabled = NO;
    _vpadnSplash = [[VpadnSplash alloc] initWithLicenseKey:@"8a80854b62d1fdc40162d205d0ff0005" target:_loadSplashView];
    _vpadnSplash.delegate = self;
    [_vpadnSplash setEndurableSecond:3];
    [_vpadnSplash loadRequest:[self initialRequest]];
}

#pragma mark -

/// 通知有廣告可供拉取 call back
- (void) onVpadnSplashLoaded:(VpadnSplash *)vpadnSplash {
    _requestButton.hidden = YES;
    [self.navigationController setNavigationBarHidden:YES animated:YES];
}

/// 通知拉取廣告失敗 call back
- (void) onVpadnSplash:(VpadnSplash *)vpadnSplash failedToLoad:(NSError *)error {
    _requestButton.enabled = YES;
}

/// 通知廣告已送出點擊事件
- (void) onVpadnSplashClicked:(VpadnSplash *)vpadnSplash {
    
}

/// 通知即將離開Application
- (void) onVpadnSplashWillLeaveApplication:(VpadnSplash *)vpadnSplash {
    
}

/// 通知廣告被允許關閉
- (void) onVpadnSplashAllowToClose:(VpadnSplash *)vpadnSplash {
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: FrameworkExample/Swift/VponSdkSampleSwift/VponSdkSampleSwift/Sdk/VponSdkSplashViewController.swift
//
//  VponSdkSplashViewController.swift
//  VponSdkSampleSwift
//
//  Created by Yi-Hsiang, Chien on 2020/2/5.
//  Copyright © 2020 EricChien. All rights reserved.
//

import UIKit

import VpadnSDKAdKit
import AdSupport

extension VponSdkSplashViewController: VpadnSplashDelegate {
    func onVpadnSplashLoaded(_ vpadnSplash: VpadnSplash) {
        self.requestButton.isHidden = true
        self.navigationController?.setNavigationBarHidden(true, animated: true)
    }
    
    func onVpadnSplash(_ vpadnSplash: VpadnSplash, failedToLoad error: Error) {
        self.requestButton.isEnabled = true
    }
    
    func onVpadnSplashAllow(toClose vpadnSplash: VpadnSplash) {
        self.navigationController?.popViewController(animated: true)
    }
    
    func onVpadnSplashWillLeaveApplication(_ vpadnSplash: VpadnSplash) {
        
    }
    
    func onVpadnSplashClicked(_ vpadnSplash: VpadnSplash) {
        
    }
}

class VponSdkSplashViewController: UIViewController {
    
    @IBOutlet weak var loadSplashView: UIView!
    
    var vpadnSplash: VpadnSplash!
    
    @IBOutlet weak var requestButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "SDK - Splash"
        requestButtonDidTouch(requestButton)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: Initial VpadnAdRequest
    
    func initialRequest() -> VpadnAdRequest {
        let request = VpadnAdRequest.init()
        request.setTestDevices([ASIdentifierManager.shared().advertisingIdentifier.uuidString])     //取得測試廣告
        request.setUserInfoGender(.genderMale)                                                      //性別
        request.setUserInfoBirthdayWithYear(2000, month: 08, andDay: 17)                            //生日
        request.setMaxAdContentRating(.general)                                                     //最高可投放的年齡(分類)限制
        request.setTagForUnderAgeOfConsent(.false)                                                  //是否專為特定年齡投放
        request.setTagForChildDirectedTreatment(.false)                                             //是否專為兒童投放
        request.addKeyword("keywordA")                                                              //關鍵字
        request.addKeyword("key1:value1")                                                           //鍵值
        return request
    }
    
    // MARK: Button Method
    
    @IBAction func requestButtonDidTouch(_ sender: UIButton) {
        sender.isEnabled = false
        
        vpadnSplash = VpadnSplash.init(licenseKey: "8a80854b62d1fdc40162d205d0ff0005", target: loadSplashView)
        vpadnSplash.delegate = self
        vpadnSplash.setEndurableSecond(3)
        vpadnSplash.load(initialRequest())
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
